torch/distributed/_functional_collectives.py

```diff
@@ -113,6 +113,19 @@ def _wait_tensor(tensor: torch.Tensor) -> torch.Tensor:
         _wait_and_clear_tensor(data_ptr, version_and_work[0])
     return tensor
 
+class WaitHolder:
+    def __init__(self, lookup_tensor):
```

Comment:
nit: can you add some documentation too why we need this `WaitHolder`? 

Also, would this `WaitHolder` handle tensor lists directly?

Reply:
WaitHolder only handle one tensor. The next PR adds proper handling of tensor lists.

Let me document it though.

Reply:
WaitHolder makes a bit more sense in the the PR once we introduce ref counting to wait.